packages/grafana-ui/src/components/DataSourceSettings/types.ts

```diff
@@ -37,4 +37,6 @@ export interface HttpSettingsProps extends HttpSettingsBaseProps {
   sigV4AuthToggleEnabled?: boolean;
   /** Azure authentication settings **/
   azureAuthSettings?: AzureAuthSettings;
+  /** If SIGV4 is enabled, provide an editor for SIGV4 connection config  **/
+  renderSigV4Editor?: React.ReactNode;
```

Comment:
only thought is if we should name this a bit more generic, customAuthEditor ?

Reply:
Would be nice with a generic handling for custom auth editors, but currently this is very explicit since there's another prop for the azure editor and they both have hard coded labels.

Reply:
@sunker what do you mean by hardcoded labels? where? Can't that be moved into the render callback?

Reply:
ah I guess you mean https://github.com/grafana/grafana/blob/e699780c6bd7be63f4f60ba1c3b968b6aa565a0c/packages/grafana-ui/src/components/DataSourceSettings/DataSourceHttpSettings.tsx#L258